features/step_definitions/redmine_steps.rb

```diff
@@ -7,59 +7,68 @@
 end
 
 Then(/^New user is registered$/) do
-  success_message = find_element_by_id('flash_notice')
 
-  expect(@driver.current_url).to eql @my_account_page_url
-  expect(success_message).to be_displayed
+
+  success_message = on(MyAccountPage).success_message_element
+
+  expect(@current_page.page_url_value).to eql on(MyAccountPage).page_url_value
```

Comment:
page_url_value will ask url value from instance of a class of the page but not from browser so you compare here the url string with itself actually )